### PR TITLE
Optimize docs

### LINKED ISSUE #1498: Revising document structure
**What would you like to be added/modified**:

**Why is this needed**:
Existing document structure is not clean, we need to split user guide and developer guide, and relocate other content, to make it easier for reading.

relavent issues: #1334, #1340, #1119, #517, #944

**Tentantive content location**
```markdown
## user guide
  - one click install
  - install with keadm
  - running with managed k8s
  - play with a demo app (something like istio playbook)


## developer guide
  - build from code
  - trouble shooting
	

## concepts
  - arch
  - components
  - api objects
    - device model / device instance


## Reference (user manual, specifications)
  - ComponentConfigAPI
  - DeviceCRD
  - mapper design/develop specifications


## community
  - resources
    - meeting
    - notes
    - videos
    - mailing list, slack
  	
  - governance
    - membership
  	
  - release management
```

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Xiang Dai <long0dai@foxmail.com>

**What type of PR is this?**
/kind documentation

**Which issue(s) this PR fixes**:

Ref: #1498

### CODE REVIEW

**Comment by daixiang0** [comment]
@kevin-wangzefeng done.

**Comment by kevin-wangzefeng** [comment]
/approve
Looks good to me for initial improvement.

File: docs/configuration/cri.md [hunk 1]
```diff
@@ -177,4 +177,55 @@ root      3941  3.0  1.0 2971576 174648 ?      Sl   17:38   0:02 /usr/bin/qemu-s
 crictl pods
 POD ID              CREATED              STATE               NAME                NAMESPACE           ATTEMPT
 b1c0911644cb9       About a minute ago   Ready               nginx-untrusted     default             0
-```
```
Comment (chendave) [inline docs/configuration/cri.md:180]: recover this line pls.

File: docs/configuration/cri.md [hunk 2]
```diff
@@ -177,4 +177,55 @@ root      3941  3.0  1.0 2971576 174648 ?      Sl   17:38   0:02 /usr/bin/qemu-s
...
+modules```
```
Comment (chendave) [inline docs/configuration/cri.md:180]: remove this line pls.

File: docs/contributing/build.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+All binaries would locate at `$GOPATH/bin`.
+
+## Build through Make
+
+You can build through `make all` after download the repository:
+
+```shell
+$ make all HELP=y
```
Comment (chendave) [inline docs/contributing/build.md:21]: I think "make all" is good enough here.

File: docs/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# Getting started
+
+KubeEdge is an open source system for extending native containerized application orchestration capabilities to hosts at Edge.
+
+In this quick-start guide, we will explain:
+
+- How to ask questions, build and contribute to Thanos.
```
Comment (chendave) [inline docs/getting-started.md:7]: Thanos?

**Comment by kadisi** [comment]
/lgtm

thanks @daixiang0

**Pull request merged**